deepinv/utils/plotting.py

```diff
@@ -115,27 +115,79 @@ def prepare_images(x=None, y=None, x_net=None, x_nl=None, rescale_mode="min_max"
     return imgs, titles, grid_image, caption
 
 
-def preprocess_img(im, rescale_mode="min_max"):
+def _preprocess_img(
+    im: torch.Tensor, rescale_only: bool, rescale_mode: str = "min_max"
+):
     r"""
     Preprocesses an image tensor for plotting.
 
     :param torch.Tensor im: the image to preprocess.
+    :param bool rescale_only: if True, only rescales the image without applying the modulus function if the input is complex-valued and without casting the image values to float32.
     :param str rescale_mode: the rescale mode, either 'min_max' or 'clip'.
     :return: the preprocessed image.
     """
-    with torch.no_grad():
-        if im.shape[1] == 2:  # for complex images
-            pimg = im.pow(2).sum(dim=1, keepdim=True).sqrt().type(torch.float32)
-        elif im.shape[1] > 3:
-            pimg = im.type(torch.float32)
-        else:
-            if torch.is_complex(im):
-                pimg = im.abs().type(torch.float32)
-            else:
-                pimg = im.type(torch.float32)
+    if not rescale_only:
+        # Apply the modulus function on each entry if the input is complex-valued
+        # or if it has 2 channels which are assume to represent the real and
+        # imaginary parts of a complex-valued image.
+        # NOTE: Expect a batched input
+        channels_dim = 1
+        n_channels = im.shape[channels_dim]
+        is_complex = torch.is_complex(im) or n_channels == 2
+        if is_complex:
+            # NOTE: This implementation is meant to work for complex-valued images
+            # and for real-valued images with two channels, understood as the real
+            # and imaginary part of a complex-valued image.
+            im = torch.linalg.vector_norm(im, ord=2, dim=channels_dim, keepdim=True)
```

Comment:
We already have this function `deepinv.loss.metric.functional.complex_abs` https://github.com/deepinv/deepinv/blob/7116f703c115bad391ab83f6b9fa6f5ccc85c75d/deepinv/loss/metric/functional.py#L39 do you think you could unify this line and this function?